server.js

```diff
@@ -1,27 +1,117 @@
 import express from "express";
 import cors from "cors";
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
+
+import netflixData from "./data/netflix-titles.json";
 // import topMusicData from "./data/top-music.json";
 
 // Defines the port the app will run on. Defaults to 8080, but can be overridden
 // when starting the server. Example command to overwrite PORT env variable value:
 // PORT=9000 npm start
+
+
 const port = process.env.PORT || 8080;
 const app = express();
-
+const listEndpoints = require('express-list-endpoints');
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.json(listEndpoints(app));
+});
+app.get('/titles', (req, res) => {
+  res.json(netflixData)
+  } )
+
+
+app.get('/title/:id', (req, res) => {
+  const { id } = req.params;
+  const showId = netflixData.filter((title) => {
+    return title.show_id === Number(id);
+  });
+  if (showId.length) {
+    res.status(200).json({
+      success: true,
+      message: "OK",
+      body: {
+        title: showId
+      }
+    });
+  } else {
+    res.status(404).json({
+      success: false,
+      message: "title not found",
+      body: {}
+    });
+  }
+});
+app.get('/title/:title', (req, res) => {
+  const { title } = req.params;
+  const netflixTitle = netflixData.filter((item) => {
+    return item.title.toLowerCase().includes(title.toLowerCase());
+  });
+  if (netflixTitle.length) {
+    res.status(200).json({
+      success: true,
+      message: "OK",
+      body: {
+        title: netflixTitle
+      }
+    });
+  } else {
+    res.status(404).json({
+      success: false,
+      message: "title not found",
+      body: {}
+    });
+  }
+});
```

Comment:
the endpoint title/:title is not working because it gets mixed up with title/:id. Because id is read first in the code the title is treated as an id and because it's not a number it runs the 404 error for the GET '/title/:id'. To solve this you can change the endpoints to be unique like, title/:title and just id/:id or use queries like title?title=Batman&id=12345678